src/main.py

```diff
@@ -161,16 +172,48 @@ def open_pr(subject, body, branch):
     base = "master"
     head = "{}:{}".format(repo.owner.login, branch)
     pr_message = ((body or "") + "\n\n" + DISCLAIMER).strip()
-    # Include closed PRs – if the maintainer has closed our last PR, we don't want to
-    # open another one.
-    for pr in origin_repo.get_pulls(state="all", base=base, head=head):
+
+    if os.path.isfile("flathub.json"):
+        with open("flathub.json") as f:
+            repocfg = json.load(f)
+            automerge = repocfg.get("automerge-flathubbot-prs")
+
+    # Enable automatic merge if external data is broken unless explicitly disabled
+    if automerge is not False and manifest_checker:
+        for data in manifest_checker.get_outdated_external_data():
+            if data.new_version and data.state == ExternalData.State.BROKEN:
+                automerge = True
+                break
+
+    prs = origin_repo.get_pulls(state="all", base=base, head=head)
+
+    # If the maintainer has closed our last PR or it was merged,
+    # we don't want to open another one.
+    closed_prs = [pr for pr in prs if pr.state == "closed"]
+    for pr in closed_prs:
         log.info(
             "Found existing %s PR: %s",
             "merged" if pr.is_merged() else pr.state,
             pr.html_url,
         )
         return
 
+    if origin_repo.permissions.push:
+        open_prs = [pr for pr in prs if pr.state == "open"]
```

Comment:
By moving the check for open PRs into this block, I think the behaviour for repos where the script _can't_ push to the origin repo will be changed: if a PR is open for the branch, it will still try and fail to create another PR for the same branch.

I think the behaviour we want, logically, is:

* if there is at least one PR for this branch:
  - if the source is currently broken, and the PR is open, and CI passed, and automerge is enabled, and it can be merged, then merge it
  - otherwise, return
* otherwise, push the branch and open a PR

Reply:
As GitHub very helpfully says `outdated` and I don't remember what the code was doing before it was labeled so, the current behavior is:

1. Check if there's newer data available.
2. Check if old data is 
3. If both conditions are met, enable automerge.

Then:

- Find open PRs.
- Check if have permissions to push and automerge is enabled.
  - Return if not.
  - Check if CI passed for PR and is mergeable.
    - Merge and return.

So if I understand correctly, the remaining issue here is that the function should return if found an open PR but have no push permissions.

Reply:
Ah, yes, I can't read. That's already the case.